release/ray_release/scripts/ray_bisect.py

```diff
@@ -102,15 +106,19 @@ def _sanity_check(test: Test, passing_revision: str, failing_revision: str) -> b
         f"Sanity check passing revision: {passing_revision}"
         f" and failing revision: {failing_revision}"
     )
-    outcomes = _run_test(test, [passing_revision, failing_revision])
-    return (
-        outcomes[passing_revision][0] == "passed"
-        and outcomes[failing_revision][0] != "passed"
+    outcomes = _run_test(test, [passing_revision, failing_revision], run_per_commit)
+    passed_on_passing_revision = all(
+        outcome == "passed" for outcome in outcomes[passing_revision].values()
```

Comment:
(nit) maybe:

````
if not all(map(lambda x: x == "passed", outcomes[passing_revision].values())):
    return False
return all(map(lambda x: x != "passed", outcomes[failing_revision].values())):
````

which is:
- slightly shorter (well, you can also achieve the same thing by renameing `outcome` to `x` I guess)
- shortcut if the first condition is False, without calculating the second one

Reply:
fun fact, the existing code is generated by github copilot ;))